containerd-shim-v2/service.go

```diff
@@ -403,7 +403,30 @@ func (s *service) Update(ctx context.Context, r *taskAPI.UpdateTaskRequest) (*pt
 
 // Wait for a process to exit
 func (s *service) Wait(ctx context.Context, r *taskAPI.WaitRequest) (*taskAPI.WaitResponse, error) {
-	return nil, errdefs.ErrNotImplemented
+	var ret uint32
+
+	s.Lock()
+	c, err := s.getContainer(r.ID)
+	s.Unlock()
```

Comment:
Don't we want to use the same global locking strategy for this one?
```go
s.Lock()
defer s.Unlock()

c, err := s.getContainer(r.ID)

...
```

Reply:
No, here we cannot use a global lock. Since there would be multiple wait service requested in a pod, such as multi-containers  or execs run in this pod. The global lock would make the wait serialized, this isn't what we want.

Reply:
Oh yeah good point, this would end up in a deadlock.